src/adoptawidget/index.md

```diff
@@ -0,0 +1,269 @@
+```
+
+### Adding more explanation to the documentation {#adding-more-explanation-to-the-documentation}
+
+If a widget doesn't provide enough explanation, you can add more! You can
+explain:
+
+*   What it is
+*   What it enables you to do
+*   How to use it
+*   The behavior of the widget in certain situations
+*   Other important information about the widget
+
+See the [Documentation section](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#documentation-dartdocs-javadocs-etc) of the Flutter Style Guide and [Effective Dart: Documentation](https://dart.dev/guides/language/effective-dart/documentation) for details on how to write great documentation.
+
+## Quickstart {#quickstart}
+
+### Get the code {#get-the-code}
+
+To get started, fork the SDK and use Git to fetch a local copy:
+
+1. Install dependencies listed in [Setting up the Framework development environment](https://github.com/flutter/flutter/wiki/Setting-up-the-Framework-development-environment)
+2. Fork the flutter repo [https://github.com/flutter/flutter](https://github.com/flutter/flutter). For more information, see [GitHub's official instructions](https://docs.github.com/en/free-pro-team@latest/github/getting-started-with-github/fork-a-repo).
+3. Clone your fork: `git clone https://github.com/&lt;your_name_here>/flutter.git`
+4. Change directory: `cd flutter`
+5. Add an upstream remote: `git remote add upstream git@github.com:flutter/flutter.git`
+6. Run `flutter update-packages` to fetch all the Dart packages Flutter uses.
+
+
+### Open the code in your IDE {#open-the-code-in-your-ide}
+
+In IntelliJ or VSCode, open `packages/flutter`, where most of the Flutter
+framework code lives.
+
+
+### Make the change {#make-the-change}
+
+Open the file listed in the issue and update the documentation comment. See
+"Ways to contribute" for examples.
+
+### Analyzing and Generating API Docs
+
+Running this command from the root directory will analyze and generate the
+documentation. This command can take up to 20 minutes to complete, so double
+check your changes first. If you want, you can also [analyze
+snippets](https://github.com/flutter/flutter/tree/master/dev/snippets#running-sample-analysis-locally)
+directly.
+
+```bash
+./dev/bots/docs.sh
+```
+
+To view the generated docs, open `dev/docs/doc/api/index.html`.
+
+**Note:** Flutter API docs cannot be generated on Windows. If you are using
+*Windows, you may need to download the Linux version of the SDK and use the
+*[Windows Subsystem for
+*Linux](https://docs.microsoft.com/en-us/windows/wsl/about).
+
+For more information, see the [Dartdoc Generation
+README](https://github.com/flutter/flutter/tree/master/dev/snippets).
+
+### Commit the change {#commit-the-change}
+
+Modify the documentation and commit the change:
+
+
+```bash
+git checkout -b update-builder-widget
+git add -A
+git commit -m "Update Builder API Docs"
+git push origin update-builder-widget
```

Comment:
Wow, you do this very differently than I do. I use vim.

Reply:
I use vim too, but the `-m` flag is simpler to create instructions for.